### PR TITLE
Refactor to assist in adding calculated summary page

### PR DESCRIPTION
**Pull request by SamGodwin2**
### What is the context of this PR?
As part of implementing the new page type of calculated summary page it was helpful to refactor the existing question page structure into a less rigid form allowing us to have multiple page types, not just question pages. As the PR with the refactor and CalSum page implementation is a bit to large I've split the wok into two parts the first of which is this refactor. 

### How to review 
All functionality should remain identical coverage should not go down and all existing tests should pass.

### CODE REVIEW

**Comment by SamGodwin2** [comment]
@pricea1 the locally failing test seems to happen for me in master as well so for now I'll just leave it alone as it's not blocking this and then tackle it in a different PR.

**Comment by SamGodwin2** [comment]
[Calculated summary (12)](https://trello.com/c/IjLzPdVy/704-calculated-summary-12)

**Comment by SamGodwin2** [comment]
The code cov drop is occurring as a large number of file were moved around some of which were missing a few lines of coverage. This means that the relative difference in coverage has gone down by around 0.3% however the overall coverage has gone up. 
If everyone else is happy I'd like to merge this without the cov passing as looking at the reports there aren't any lines that have become uncovered in this pr.

File: eq-author/src/App/page/Design/EditorLayout/Tabs/index.js [hunk 1]
```diff
@@ -92,7 +92,7 @@ export const UnwrappedTabs = props => {
                 Component: Tab,
                 otherProps: { to: url(match), activeClassName },
               }
-            : { Component: DisabledTab, otherProps: {} };
+            : { Component: DisabledTab, otherProps: { disabled: true } };
```
Comment (tgandrews) [inline eq-author/src/App/page/Design/EditorLayout/Tabs/index.js]: Surely a disabled tab should be disabled by default?

Comment (tgandrews) [reply eq-author/src/App/page/Design/EditorLayout/Tabs/index.js]: I just looked into this. This prop is completely unnecessary. The DisabledTab is a span. Can we remove this and change the cypress test so that it checks that it becomes a link not loses the disabled attribute which wasn't doing anything.

File: eq-author/src/App/page/Design/PageHeader.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    );
+  };
+
+  render() {
+    const {
+      page,
+      onChange,
+      onUpdate,
+      match,
+      moveDisabled,
```
Comment (tgandrews) [inline eq-author/src/App/page/Design/PageHeader.js]: Could we prefix this and the one below with `is` so that its obvious that they are flags and not a way of moving the disabled button.

File: eq-author/src/App/page/Design/QuestionPageEditor/index.js [hunk 3]
```diff
@@ -3,21 +3,29 @@ import styled from "styled-components";
...
+import withPropRenamed from "enhancers/withPropRenamed";
+import PageHeader from "../PageHeader";
 
 import AnswersEditor from "./AnswersEditor";
 import AnswerTypeSelector from "./AnswerTypeSelector";
-import MovePageModal from "./MovePageModal";
-import MovePageQuery from "./MovePageModal/MovePageQuery";
+import withUpdateQuestionPage from "./withUpdateQuestionPage";
+
+import withUpdateAnswer from "../answers/withUpdateAnswer";
```
Comment (tgandrews) [inline eq-author/src/App/page/Design/QuestionPageEditor/index.js]: Could we move these next to this component or are they shared?

File: eq-author/src/App/page/Design/Validation/MetadataContentPicker.js [hunk 4]
```diff
@@ -5,7 +5,7 @@ import { get } from "lodash";
 import ContentPickerSelect from "components/ContentPickerSelect";
 import { METADATA } from "components/ContentPickerSelect/content-types";
 
-import AvailableMetadataQuery from "App/questionPage/Design/Validation/AvailableMetadataQuery";
+import AvailableMetadataQuery from "App/page/Design/Validation/AvailableMetadataQuery";
```
Comment (tgandrews) [inline eq-author/src/App/page/Design/Validation/MetadataContentPicker.js]: I think this could be 
```suggestion
import AvailableMetadataQuery from "./AvailableMetadataQuery";
```

File: eq-author/src/App/page/Design/Validation/FieldWithInclude.test.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import React from "react";
 import { shallow } from "enzyme";
-import FieldWithInclude from "App/questionPage/Design/Validation/FieldWithInclude";
+import FieldWithInclude from "App/page/Design/Validation/FieldWithInclude";
```
Comment (tgandrews) [inline eq-author/src/App/page/Design/Validation/FieldWithInclude.test.js]: I think this can be
```suggestion
import FieldWithInclude from "./FieldWithInclude";
```

File: eq-author/src/App/page/Design/Validation/PreviousAnswerContentPicker.js [hunk 6]
```diff
@@ -6,7 +6,7 @@ import ContentPickerSelect from "components/ContentPickerSelect/index";
 import { ANSWER } from "components/ContentPickerSelect/content-types";
 
 import shapeTree from "components/ContentPicker/shapeTree";
-import AvailablePreviousAnswersQuery from "App/questionPage/Design/Validation/AvailablePreviousAnswersQuery";
+import AvailablePreviousAnswersQuery from "App/page/Design/Validation/AvailablePreviousAnswersQuery";
```
Comment (tgandrews) [inline eq-author/src/App/page/Design/Validation/PreviousAnswerContentPicker.js]:
```suggestion
import AvailablePreviousAnswersQuery from "./AvailablePreviousAnswersQuery";
```

File: eq-author/src/App/page/Design/Validation/PreviousAnswerContentPicker.test.js [hunk 7]
```diff
@@ -1,6 +1,6 @@
 import React from "react";
 import { shallow } from "enzyme";
-import { UnwrappedPreviousAnswerContentPicker } from "App/questionPage/Design/Validation/PreviousAnswerContentPicker";
+import { UnwrappedPreviousAnswerContentPicker } from "App/page/Design/Validation/PreviousAnswerContentPicker";
```
Comment (tgandrews) [inline eq-author/src/App/page/Design/Validation/PreviousAnswerContentPicker.test.js]:
```suggestion
import { UnwrappedPreviousAnswerContentPicker } from "./PreviousAnswerContentPicker";
```

File: eq-author/src/App/page/Design/Validation/ValidationPills.test.js [hunk 8]
```diff
@@ -5,7 +5,7 @@ import { omit } from "lodash";
 import {
   Pills,
   ValidationPills,
-} from "App/questionPage/Design/Validation/ValidationPills";
+} from "App/page/Design/Validation/ValidationPills";
```
Comment (tgandrews) [inline eq-author/src/App/page/Design/Validation/ValidationPills.test.js]:
```suggestion
} from "./ValidationPills";
```

File: eq-author/src/App/page/Design/Validation/ValidationView.test.js [hunk 9]
```diff
@@ -1,6 +1,6 @@
 import React from "react";
 import { shallow } from "enzyme";
-import ValidationView from "App/questionPage/Design/Validation/ValidationView";
+import ValidationView from "App/page/Design/Validation/ValidationView";
```
Comment (tgandrews) [inline eq-author/src/App/page/Design/Validation/ValidationView.test.js]:
```suggestion
import ValidationView from "./ValidationView";
```

File: eq-author/src/App/page/Design/Validation/withToggleAnswerValidation.test.js [hunk 10]
```diff
@@ -1,4 +1,4 @@
-import { mapMutateToProps } from "App/questionPage/Design/Validation/withToggleAnswerValidation";
+import { mapMutateToProps } from "App/page/Design/Validation/withToggleAnswerValidation";
```
Comment (tgandrews) [inline eq-author/src/App/page/Design/Validation/withToggleAnswerValidation.test.js]:
```suggestion
import { mapMutateToProps } from ".withToggleAnswerValidation";
```

File: eq-author/src/App/page/Design/Validation/withUpdateAnswerValidation.test.js [hunk 11]
```diff
@@ -1,4 +1,4 @@
-import { mapMutateToProps } from "App/questionPage/Design/Validation/withUpdateAnswerValidation";
+import { mapMutateToProps } from "App/page/Design/Validation/withUpdateAnswerValidation";
```
Comment (tgandrews) [inline eq-author/src/App/page/Design/Validation/withUpdateAnswerValidation.test.js]:
```suggestion
import { mapMutateToProps } from "./withUpdateAnswerValidation";
```

File: eq-publisher/src/getQuestionnaire/queries.js [hunk 12]
```diff
@@ -10,7 +10,7 @@ exports.getQuestionnaire = `
     qCode
     ...BasicAnswer
   }
-
+
```
Comment (tgandrews) [inline eq-publisher/src/getQuestionnaire/queries.js]: Looks like unnecessary whitespace changes in this file.

File: eq-author/src/enhancers/withPropRenamed.js [hunk 13]
```diff
@@ -1,9 +1,10 @@
 import React from "react";
 
-export default (oldName, newName) => Component => props => {
-  const newProps = {
-    [newName]: props[oldName],
-    ...props,
+export default (oldName, newName) => WrappedComponent =>
+  class RemappedComponent extends React.Component {
```
Comment (tgandrews) [inline eq-author/src/enhancers/withPropRenamed.js]: You can set the displayName of the component to be something like `withPropRenamed(${WrappedComponent.displayName})` so that it gets that name not this one.

File: eq-author/src/App/page/Design/QuestionPageEditor/index.js (hunk 3 above)
Comment (SamGodwin2) [reply eq-author/src/App/page/Design/QuestionPageEditor/index.js]: Looks like the answerUpdate is shared between this and the properties panel.
